### ISSUE TITLE
"astralsorcery" NBT tags interfere with TConstruct casting

### ISSUE THREAD

**Issue by codewarrior0** [HellFirePvP/AstralSorcery#636]
AstralSorcery built from today's `master` at commit d06a67a1054037e8cd0d347de4c40f856571a77d

Tool parts from Tinker's Construct cannot be used to create casts after Astral Sorcery has added its `"astralsorcery":{}` NBT tag to them. This is due to the casting table recipe matcher requiring an exact NBT match for recipe inputs, on account of TConstruct items and tool parts using NBT tags to indicate the shape of a cast and the material of a tool part.

This tag is being added as a result of the ASM hack which adds extra enchantment levels according to an amulet-enchantment mechanic, which was introduced in f3f8c3966ad55cda6ad15b899314c695e7880617.

**Comment by HellFirePvP** [comment]
Congratulations, you found a bug in an unreleased build.
Guess what: That's one of the reasons it's not released. It's not finished and not tested. Things are broken and things don't work properly. That's why you don't build from source and report bugs about it. It is already long fixed on my local dev instance, i didn't bother to push it yet - many people like to build mods from unreleased things from github nowadays for whatever reason and think everything that's pushed is stable and finished. 

tl;dr: Don't report bugs on unfinished things.

On the other hand, strict NBT matching is horrible as it is and shouldn't be done anymore anyway. There are a multitude of possible approaches to matching an NBT tag that are different than .equals and serve a better and more dynamic approach to figuring out if the stack has the value the mod in question is looking for. Expecting an NBT-Tag to not be modified by other mods is not more than wishful thinking.

**Comment by codewarrior0** [comment]
I'm sorry if I've upset you. I chose to report this particular issue early because it will have lasting effects on the game world that persist even after Astral Sorcery is rolled back or removed.

Looking forward to your next push.
